docs/source/examples/auto-failover.rst

```diff
@@ -76,41 +70,112 @@ AWS, where it succeeded after two regions:
   I 02-23 16:47:04 cloud_vm_ray_backend.py:740] Successfully provisioned or found existing VM. Setup completed.
+
+
+Multiple Candidate GPUs
+-------------------------
+
+If a task can be run on different GPUs, the user can specific multiple candidate GPUs,
+and SkyPilot will automatically find the best available GPU.
+
+To allow SkyPilot to choose any of the candidate GPUs, specify a set of candidate GPUs in the task yaml:
+
+.. code-block:: yaml
+
+  resources:
+    accelerators: {A10:1, L4:1, A10g:1}
+
+In the above example, SkyPilot will try to provision the any cheapest available GPU within the set of
+A10, L4, and A10g GPUs, using :code:`sky launch task.yaml`.
+
+.. code-block:: console
+
+  $ sky launch task.yaml
+  ...
+  -----------------------------------------------------------------------------------------------------
+  I 11-19 08:07:45 optimizer.py:910]  CLOUD   INSTANCE                 vCPUs   Mem(GB)   ACCELERATORS   REGION/ZONE   COST ($)   CHOSEN   
+  I 11-19 08:07:45 optimizer.py:910] -----------------------------------------------------------------------------------------------------
+  I 11-19 08:07:45 optimizer.py:910]  Azure   Standard_NV6ads_A10_v5   6       55        A10:1          eastus        0.45          ✔     
+  I 11-19 08:07:45 optimizer.py:910]  GCP     g2-standard-4            4       16        L4:1           us-east4-a    0.70                
+  I 11-19 08:07:45 optimizer.py:910]  AWS     g5.xlarge                4       16        A10G:1         us-east-1     1.01                
+  I 11-19 08:07:45 optimizer.py:910] -----------------------------------------------------------------------------------------------------
+
+
+
+To specify a preference order, use a list of candidate GPUs in the task yaml:
+
+.. code-block:: yaml
+
+  resources:
+    accelerators: [A10:1, A10g:1, L4:1]
+
+In the above example, SkyPilot will first try to provision an A10 GPU, then an A10g GPU, and finally an L4 GPU:
+:code:`sky launch task.yaml`
+
+
+(**Advanced**) Multiple Candidate Resources
+--------------------------------------------
+
+If a task would like to specify multiple candidate resources (not only GPUs), the user can specify a list of candidate resources with a preference annotation:
+
+.. code-block:: yaml
+
+  resources:
+    ordered: # Candidate resources in a preference order
+      - cloud: gcp
+        accelerators: A100-80GB
+      - instance_type: g5.xlarge
+      - cloud: azure
+        region: eastus
+        accelerator: A100
+
+
+
+.. code-block:: yaml
+
+    resources:
+      any_of: # Candidate resources that can be chosen in any order
+        - cloud: gcp
+          accelerators: A100-80GB
+        - instance_type: g5.xlarge
+        - cloud: azure
+          region: eastus
+          accelerator: A100
+          
+This will be useful when only a set of regions/clouds are desired for launching the resources:
+
+.. code-block:: yaml
+
+  resources:
+    any_of:
+      - region: us-east-1
```

Comment:
This example is a bit complicated. Some regions are in AWS and some in GCP. Maybe simplify the example a bit

Reply:
Just reduced the candidate to only two. PTAL : )